src/mesh/mesh.cpp

```diff
@@ -582,14 +582,14 @@ Mesh::Mesh(ParameterInput *pin, ApplicationInput *app_in, RestartReader &rr,
   // the file is already open and the pointer is set to after <par_end>
 
   // All ranks read HDF file
-  nbnew = rr.GetAttr<int>("Mesh", "nbnew");
-  nbdel = rr.GetAttr<int>("Mesh", "nbdel");
-  nbtotal = rr.GetAttr<int>("Mesh", "nbtotal");
-  root_level = rr.GetAttr<int32_t>("Mesh", "rootLevel");
+  nbnew = rr.GetAttr<int32_t>("Info", "NBNew");
+  nbdel = rr.GetAttr<int32_t>("Info", "NBDel");
+  nbtotal = rr.GetAttr<int32_t>("Info", "NumMeshBlocks");
+  root_level = rr.GetAttr<int32_t>("Info", "RootLevel");
```

Comment:
Yeah, I'm not super happy with this. Using `int32_t` here instead of `int` is basically being overly cautious in case we run on platforms where `int != int32_t` and we want to read and HDF5 file written on a platform where `int == int32_t` (or the other way around).

Although, I just realize that this means we'd also have to explicitly cast to `int32_t` when writing these attributes, which we currently don't do. They're declared as `int` and get written as such. So I think we should remove all references to `int32_t` instead and just use `int`. If we ever run into an issue of a platform where `int != int32_t`, we'll deal with it then.